install/helm/agones/templates/_helpers.tpl

```diff
@@ -69,4 +69,24 @@ feature gates. In the process, we validate that the feature gate is known.
 {{- end -}}
 {{- end -}}
 {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{/* 
+Derives the replica number for agones-controller based on current helm and feature flag settings.
+*/}}
+{{- define "agones.controller.replica" -}}
```

Comment:
I forgot this was landing all in `controller.yaml`, any reason we can't just inline the logic rather than adding the helper? I think if you inline it at e.g. line 16 in controller.yaml, you don't even need a `define`, it just becomes a matter of setting $controllerReplicas.